src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -510,28 +478,42 @@ void VTKPolyDataWriterInterface::writeUnstructuredGrid( vtkSmartPointer< vtkUnst
   vtuWriter->Write();
 }
 
-string VTKPolyDataWriterInterface::getTimeStepSubFolder( real64 time ) const
+string VTKPolyDataWriterInterface::getTimeStepSubFolder( real64 const time ) const
 {
-  return m_outputFolder + "/" + std::to_string( time );
+  return joinPath( m_outputName, std::to_string( time ) );
 }
 
-void VTKPolyDataWriterInterface::write( real64 time, integer cycle, DomainPartition const & domain )
+void VTKPolyDataWriterInterface::write( real64 const time,
+                                        integer const cycle,
+                                        DomainPartition const & domain )
 {
-  createTimeStepSubFolder( time );
+  string const stepSubFolder = getTimeStepSubFolder( time );
+  if( MpiWrapper::commRank( MPI_COMM_GEOSX ) == 0 )
+  {
+    if( m_previousCycle == -1 )
+    {
+      makeDirsForPath( joinPath( m_outputDir, m_outputName ) );
+    }
```

Comment:
The VTK output folder is now created on first write as opposed to writer object creation time, because the name of the directory is set *after* reading the XML input.